Docker/Dockerfile

```diff
@@ -1,26 +1,46 @@
 FROM library/python:3.7.0-alpine3.8
 
-RUN apk add build-base
-RUN apk add portaudio-dev
-RUN apk add sqlite
-RUN pip install webob
+LABEL maintainer="istvan@codekuklin.com"
 
-COPY anki-sync-server /app/anki-sync-server
+ARG SERVER_COMMIT='8551021f2b2356e3c3449c1ea6b9745f0856af66'
+ARG ANKI_COMMIT='cca3fcb2418880d0430a5c5c2e6b81ba260065b7'
 
-WORKDIR /app/anki-sync-server
+SHELL ["/bin/sh", "-euxo", "pipefail", "-c"]
+
+RUN  adduser ankisync -D -s /bin/false
 
-RUN \
-    cd anki-bundled && \
-    pip install -r requirements.txt
+RUN apk add --no-cache \
+        build-base \
+        portaudio-dev \
+        sqlite \
+ && pip install --no-cache-dir webob
 
-RUN mkdir /app/data && \
-    mv /app/anki-sync-server/ankisyncd.conf /app/anki-sync-server/ankisyncd.conf.example && \
-    ln -s /app/data/ankisyncd.conf /app/anki-sync-server/
+COPY ./app /app
 
-COPY config /app/config
-COPY scripts /app/scripts
+WORKDIR /app/anki-sync-server
 
-CMD /app/scripts/startup.sh
+RUN apk add --no-cache --virtual .install-deps git \
```

Comment:
Love the idea. Not the biggest fan of the implementation. This includes git in your final build. The overall plan is to just use wget or pip to install a single library that allows you to run the server. I would remove this for now and open a ticket so it can be discussed further.

Reply:
It won't. In line 36 it gets removed in the same RUN instruction / container layer. So git won't be in the final image neither in the "inbetween" layers.

Reply:
I would say it's pretty clean at the end.

Reply:
Well that's fair. I still think there's a simpler implementation using multi-stage builds, but if you're currently cleaning up your build files and dependencies that I would consider this comment resolved.